src/lib/linux-api/src/signal.rs

```diff
@@ -960,6 +994,58 @@ impl sigaction {
     const SIG_DFL: usize = 0;
     const SIG_IGN: usize = 1;
 
+    /// Consider [`sigaction::new_with_default_restorer`] instead, which takes
+    /// care of setting up a restorer.
+    pub fn new_raw(
+        handler: SignalHandler,
+        flags: SigActionFlags,
+        mask: sigset_t,
+        restorer: Option<unsafe extern "C" fn()>,
+    ) -> Self {
+        let handler = match handler {
+            SignalHandler::Handler(h) => Some(h),
+            SignalHandler::Action(a) => {
+                assert!(flags.contains(SigActionFlags::SA_SIGINFO));
+                Some(unsafe {
+                    core::mem::transmute::<
+                        unsafe extern "C" fn(i32, *mut siginfo_t, *mut core::ffi::c_void),
+                        unsafe extern "C" fn(i32),
+                    >(a)
+                })
+            }
+            SignalHandler::SigIgn => Some(unsafe {
+                core::mem::transmute::<usize, unsafe extern "C" fn(i32)>(Self::SIG_IGN)
+            }),
```

Comment:
Is it allowed by rust to transmute from a non-pointer to a `fn()`? I guess there's no other way to construct a C sigaction struct without doing this, but it would be nice to have some comments here explaining why it's believed to be safe.

Reply:
Yeah, unclear but there's no other way. Added some explanation and extra checks to help ensure it gets transmuted back to the correct type when interpreted.